### PR TITLE
esp32: Provide Signal class

### PR DESCRIPTION
**Pull request by bgamari**
Correct me if I'm wrong but I believe this is all that is necessary.

### CODE REVIEW

File: esp32/machine_pin.c [hunk 1]
```diff
@@ -335,6 +342,22 @@ STATIC mp_obj_t machine_pin_irq_trigger(size_t n_args, const mp_obj_t *args) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(machine_pin_irq_trigger_obj, 1, 2, machine_pin_irq_trigger);
 
+STATIC mp_uint_t pin_ioctl(mp_obj_t self_in, mp_uint_t request, uintptr_t arg, int *errcode) {
```
Comment (dpgeorge) [inline esp32/machine_pin.c]: Please move this function to earlier in the file so the forward declaration of it is not needed (put the function just before machine_pin_locals_dict_table).

File: esp32/machine_pin.c [hunk 2]
```diff
@@ -335,6 +342,22 @@ STATIC mp_obj_t machine_pin_irq_trigger(size_t n_args, const mp_obj_t *args) {
...
+  (void)errcode;
+  machine_pin_obj_t *self = self_in;
+
+  switch (request) {
+  case MP_PIN_READ: {
```
Comment (dpgeorge) [inline esp32/machine_pin.c]: Please follow existing conventions for indenting: 4 spaces and "case" will be indented a level.

Comment (bgamari) [reply esp32/machine_pin.c]: Ahh, I hadn't noticed the difference here. Note that this code came from the `esp8266` port, so some reformatting may be in order over there.

File: esp32/machine_pin.c (hunk 1 above)
Comment (bgamari) [reply esp32/machine_pin.c]: Alright.

**Comment by bgamari** [comment]
How does this look?

**Comment by dpgeorge** [comment]
Thank you!  Merged in 79feb95670311eee5ffb61f1dfd8183901d5e441

**Pull request closed without merging**